examples/07_join_aggregation.py

```diff
@@ -0,0 +1,438 @@
+ratings["day_name"] = ratings["timestamp"].dt.day_name()
+
+
+# ## Quick Data Exploration
+
+# In[3]:
+
+
+from matplotlib import pyplot as plt
+import seaborn as sns
+
+sns.set_style("darkgrid")
+
+
+def make_barplot(x, y, title):
+    norm = plt.Normalize(y.min(), y.max())
+    cmap = plt.get_cmap("magma")
+
+    ax = sns.barplot(
+        x=x,
+        y=y,
+        palette=cmap(norm(y)),
+    )
+    plt.xticks(rotation=30)
+    plt.xlabel(None)
+    plt.ylabel(None)
+    plt.tight_layout()
+    plt.title(title)
+
+
+# In[4]:
+
+
+year_volume = ratings["year"].value_counts().sort_index()
+
+make_barplot(
+    x=year_volume.index,
+    y=year_volume.values,
+    title="Yearly volume of ratings",
+)
+
+
+# In[5]:
+
+
+ratings["day_index"] = ratings["timestamp"].dt.weekday
+
+day_name_index = (
+    ratings[["day_index", "day_name"]].drop_duplicates().set_index("day_index")
+)
+day_name_index.sort_index()
+
+
+# In[6]:
+
+
+daily_volume = (
+    ratings["day_index"].value_counts().sort_index().to_frame().join(day_name_index)
+)
+
+make_barplot(
+    x=daily_volume["day_name"], y=daily_volume["count"], title="Daily volume of ratings"
+)
+
+
+# In[7]:
+
+
+user_count_ratings = (
+    ratings.groupby("userId").agg(n_ratings=("rating", "count")).reset_index()
+)
+
+print(
+    "min:",
+    user_count_ratings["n_ratings"].min(),
+    "max:",
+    user_count_ratings["n_ratings"].max(),
+)
+
+ax = sns.histplot(user_count_ratings["n_ratings"])
+ax.set(
+    xlabel="Number of ratings given by users",
+    ylabel="Total number of users",
+    title="Number of ratings given by user",
+)
+
+
+# In[8]:
+
+
+rating_count = ratings["rating"].value_counts().sort_index()
+
+make_barplot(
+    x=rating_count.index,
+    y=rating_count.values,
+    title="Distribution of ratings given to movies",
+)
+
+
+# In[9]:
+
+
+## Split train test by date
+
+min_date = ratings["timestamp"].min()
+max_date = ratings["timestamp"].max()
+duration = max_date - min_date
+train_threshold = max_date - duration / 3
+
+train = ratings.loc[ratings["timestamp"] < train_threshold]
+test = ratings.loc[ratings["timestamp"] >= train_threshold]
+train.shape, test.shape
+
+
+# ## Feature Engineering
+
+# ***Ohe Hot Encoding day names***
+
+# In[10]:
+
+
+from sklearn.pipeline import make_pipeline, FunctionTransformer
+from sklearn.compose import make_column_transformer
+from sklearn.preprocessing import OneHotEncoder
+
+col_trans = make_column_transformer(
+    ("passthrough", ["userId", "movieId", "rating"]),
+    (OneHotEncoder(sparse_output=False), ["day_name"]),
+    remainder="drop",
+    verbose_feature_names_out=False,
+)
+
+train = col_trans.fit_transform(train)
+train = pd.DataFrame(train, columns=col_trans.get_feature_names_out())
+train.head()
+
+
+# ***Join aggregator on `movieId` and `userId`***
+
+# In[11]:
+
+
+from skrub import JoinAggregator
+
+cols_day_name = list(set(train.columns) - set(["userId", "movieId"]))
+
+join_agg_rating = JoinAggregator(
```

Comment:
Should we simplify the API by accepting only one external table?

(open question: I'm shuffling ideas around to make things more readable.

Reply:
I find it hard to understand as well, I think it would be good to simplify the syntax if possible. Just like we did for the fuzzy_join, it would be interesting to ask around what users think is easier (maybe during the sprint?).

Reply:
Also, for this first join-aggregation in the example, maybe it's best to do something simpler? For instance not to do both mean and sum or on multiple tables, just for better illustration purpose.

The dropping columns part below would also then disappear for now.